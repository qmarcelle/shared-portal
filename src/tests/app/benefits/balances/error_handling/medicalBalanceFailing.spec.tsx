import BalancesPage from '@/app/(common)/myplan/benefits/balances/page';
import { mockedAxios } from '@/tests/__mocks__/axios';
import { mockedFetch } from '@/tests/setup';
import { createAxiosErrorForTest, fetchRespWrapper } from '@/tests/test_utils';
import '@testing-library/jest-dom';
import { render, screen, within } from '@testing-library/react';

jest.mock('src/auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        currUsr: {
          plan: { memCk: '123456789', grpId: '87898', sbsbCk: '654567656' },
        },
        vRules: {
          dental: true,
          dentalCostsEligible: true,
          enableCostTools: true,
        },
      },
    }),
  ),
}));

describe('Medical Balances Error Handling with Dental working', () => {
  console.log(process.env.ES_PORTAL_SVCS_API_URL);
  mockedAxios.get
    // dental
    .mockResolvedValueOnce({
      data: {
        accumulatorsDetails: [
          {
            productType: 'D',
            members: [
              {
                memberCK: 54363201,
                listofSerLimitMetDetails: [
                  {
                    accumNum: 819,
                    metAmount: 0.0,
                  },
                  {
                    accumNum: 827,
                    metAmount: 0.0,
                  },
                ],
              },
              {
                memberCK: 91722407,
                listofSerLimitMetDetails: [
                  {
                    accumNum: 819,
                    metAmount: 0.0,
                  },
                ],
              },
            ],
            serviceLimitDetails: [
              {
                accumNum: 819,
                serviceDesc: '$3000 Annual Maximum Basic and Major',
                isDollarLimit: true,
                isDays: false,
                maxAllowedAmount: 3000.0,
              },
              {
                accumNum: 827,
                serviceDesc: '$3000 Ortho Lifetime Maximum',
                isDollarLimit: true,
                isDays: false,
                maxAllowedAmount: 3000.0,
              },
            ],
          },
        ],
      },
    })
    // medical
    .mockRejectedValueOnce(
      createAxiosErrorForTest({ errorObject: {}, status: 400 }),
    );
  // loggedIn userInfo for member names
  mockedFetch.mockResolvedValueOnce(
    fetchRespWrapper({
      isActive: true,
      subscriberLoggedIn: true,
      lob: 'REGL',
      groupData: {
        groupID: '100000',
        groupCK: '21908',
        groupName: 'Chris B Hall Enterprises',
        parentGroupID: '100001',
        subGroupID: '0001',
        subGroupCK: 28951,
        subGroupName: 'Chris B Hall Enterprises',
        clientID: 'EI',
        policyType: 'INT',
        groupEIN: '620427913',
      },
      networkPrefix: 'QMI',
      subscriberID: '902218823',
      subscriberCK: '91722400',
      subscriberFirstName: 'CHRIS',
      subscriberLastName: 'HALL',
      subscriberTitle: '',
      subscriberDateOfBirth: '08/06/1959',
      subscriberOriginalEffectiveDate: '01/01/2001',
      members: [
        {
          isActive: true,
          memberOrigEffDt: '06/29/2009',
          memberCk: 91722407,
          firstName: 'CHRISTMAS',
          middleInitial: '',
          lastName: 'HALL',
          title: '',
          memRelation: 'S',
          birthDate: '06/29/2009',
          gender: 'M',
          memberSuffix: 6,
          mailAddressType: 'H',
          workPhone: '',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'M',
              planID: 'MBPX0806',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Blue Network P',
            },
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: false,
          hasSocial: true,
          esipharmacyEligible: true,
        },
        {
          isActive: true,
          memberOrigEffDt: '06/15/2009',
          memberCk: 91722406,
          firstName: 'KRISSY',
          middleInitial: 'C',
          lastName: 'HALL',
          title: '',
          memRelation: 'D',
          birthDate: '06/15/2009',
          gender: 'F',
          memberSuffix: 5,
          mailAddressType: 'H',
          workPhone: '',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'M',
              planID: 'MBPX0806',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Blue Network P',
            },
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: false,
          hasSocial: false,
          esipharmacyEligible: true,
        },
        {
          isActive: true,
          memberOrigEffDt: '05/01/2009',
          memberCk: 91722405,
          firstName: 'CHRISTIAN',
          middleInitial: '',
          lastName: 'HALL',
          title: '',
          memRelation: 'S',
          birthDate: '10/31/2011',
          gender: 'M',
          memberSuffix: 4,
          mailAddressType: 'H',
          workPhone: '',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'M',
              planID: 'MBPX0806',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Blue Network P',
            },
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: false,
          hasSocial: true,
          esipharmacyEligible: true,
        },
        {
          isActive: true,
          memberOrigEffDt: '01/01/2001',
          memberCk: 91722402,
          firstName: 'KRISTY',
          middleInitial: '',
          lastName: 'HALL',
          title: '',
          memRelation: 'W',
          birthDate: '05/17/1971',
          gender: 'F',
          memberSuffix: 1,
          mailAddressType: 'H',
          workPhone: '',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: false,
          hasSocial: true,
          esipharmacyEligible: false,
        },
        {
          isActive: true,
          memberOrigEffDt: '01/01/2001',
          memberCk: 91722401,
          firstName: 'CHRIS',
          middleInitial: 'B',
          lastName: 'HALL',
          title: '',
          memRelation: 'M',
          birthDate: '08/06/1959',
          gender: 'M',
          memberSuffix: 0,
          mailAddressType: 'H',
          workPhone: '1234567890',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'D',
              coverageLevel: '*',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'D',
              planID: 'DEHCNY02',
              effectiveDate: 1509508800000,
              planStartDate: 1451624400000,
              planClassID: 'PPOA',
              networkPlanName: 'DentalBlue Preferred Network',
            },
            {
              productCategory: 'M',
              planID: 'MBPX0806',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Blue Network P',
            },
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: true,
          hasSocial: true,
          esipharmacyEligible: true,
        },
        {
          isActive: true,
          memberOrigEffDt: '01/01/2014',
          memberCk: 91722409,
          firstName: 'CHRISTOFF',
          middleInitial: '',
          lastName: 'HALL',
          title: '',
          memRelation: 'S',
          birthDate: '01/01/2000',
          gender: 'M',
          memberSuffix: 8,
          mailAddressType: 'H',
          workPhone: '',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'M',
              planID: 'MBPX0806',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Blue Network P',
            },
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: false,
          hasSocial: false,
          esipharmacyEligible: true,
        },
        {
          isActive: true,
          memberOrigEffDt: '12/11/2006',
          memberCk: 91722408,
          firstName: 'CHRISTO',
          middleInitial: '',
          lastName: 'HALL',
          title: '',
          memRelation: 'S',
          birthDate: '12/11/2006',
          gender: 'M',
          memberSuffix: 7,
          mailAddressType: 'H',
          workPhone: '',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'M',
              planID: 'MBPX0806',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Blue Network P',
            },
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: false,
          hasSocial: true,
          esipharmacyEligible: true,
        },
        {
          isActive: true,
          memberOrigEffDt: '12/11/2006',
          memberCk: 54363201,
          firstName: 'ChrisBalance',
          middleInitial: '',
          lastName: 'HALL',
          title: '',
          memRelation: 'S',
          birthDate: '12/11/2006',
          gender: 'M',
          memberSuffix: 7,
          mailAddressType: 'H',
          workPhone: '',
          otherInsurance: [],
          coverageTypes: [
            {
              productType: 'M',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'S',
              coverageLevel: 'A',
              exchange: false,
              indvGroupInd: '',
              pedAdultInd: '',
            },
            {
              productType: 'V',
              coverageLevel: 'A',
              exchange: true,
              indvGroupInd: 'Group',
              pedAdultInd: 'Adult',
            },
          ],
          planDetails: [
            {
              productCategory: 'M',
              planID: 'MBPX0806',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Blue Network P',
            },
            {
              productCategory: 'S',
              planID: 'WSXM0218',
              effectiveDate: 1546318800000,
              planStartDate: 1514782800000,
              planClassID: 'PPOA',
              networkPlanName: 'Wellness Plan',
            },
            {
              productCategory: 'V',
              planID: 'VEMGN002',
              effectiveDate: 1546318800000,
              planStartDate: 1383278400000,
              planClassID: 'PPOA',
              networkPlanName: 'Vision Blue',
            },
          ],
          inXPermissions: true,
          futureEffective: false,
          loggedIn: false,
          hasSocial: true,
          esipharmacyEligible: true,
        },
      ],
      coverageTypes: [
        {
          productType: 'M',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'S',
          coverageLevel: 'A',
          exchange: false,
          indvGroupInd: '',
          pedAdultInd: '',
        },
        {
          productType: 'V',
          coverageLevel: 'A',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
        {
          productType: 'D',
          coverageLevel: '*',
          exchange: true,
          indvGroupInd: 'Group',
          pedAdultInd: 'Adult',
        },
      ],
      addresses: [
        {
          type: '1',
          address1: 'TEST BLUE ACCESS',
          address2: '',
          address3: '',
          city: 'Chattanooga',
          state: 'TN',
          zipcode: '37412',
          county: 'HAMILTON',
          phone: '',
          email: '',
        },
        {
          type: '2',
          address1: 'TEST 2 BLUE ACCESS',
          address2: '',
          address3: '',
          city: 'Chattanooga',
          state: 'TN',
          zipcode: '37402',
          county: 'HAMILTON',
          phone: '',
          email: '',
        },
        {
          type: 'H',
          address1: '1 CAMERON HILL CIRCLE',
          address2: '',
          address3: '',
          city: 'CHATTANOOGA',
          state: 'TN',
          zipcode: '37402',
          county: 'HAMILTON',
          phone: '4235353065',
          email: '',
        },
      ],
      healthCareAccounts: [],
      esigroupNum: '100000MBPX0806',
      cmcondition: [],
    }),
  );

  it('should call Balances api and render error for Medical Balance, success for Dental', async () => {
    const { container } = render(await BalancesPage());

    // Container Headers need to be visible
    expect(screen.getByText('Medical & Pharmacy Balance')).toBeVisible();
    expect(screen.getByText('Dental Balance')).toBeVisible();

    // Medical Balance
    // Show the user names
    // 1 for dropdown selected, 1 for dropdown option
    const medicalSec = screen.getByText(
      'Medical & Pharmacy Balance',
    ).parentElement;
    expect(screen.queryAllByText('Chris HALL').length).toBe(0);
    // Error Screen to be shown
    expect(
      within(medicalSec).getByText(
        'There was a problem loading your information. Please try refreshing the page or returning to this page later.',
      ),
    ).toBeVisible();

    // Initial render
    expect(container).toMatchSnapshot();

    // Dental Balance Section
    expect(screen.getAllByText('ChrisBalance HALL').length).toBe(2);
    expect(
      screen.getAllByText('You do not have any Deductible amounts.').length,
    ).toBe(1);
    expect(
      screen.getAllByText('You do not have any Out-of-Pocket amounts.').length,
    ).toBe(1);

    // Medical, Dental Balance, LoggedInUserInfo Api calls were called
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/api/member/v1/members/bySubscriberCk/654567656/balances/deductibleAndOOP/D',
    );
    expect(mockedAxios.get).toHaveBeenCalledWith(
      '/api/member/v1/members/bySubscriberCk/654567656/balances/deductibleAndOOP/M',
    );
    expect(mockedFetch).toHaveBeenCalledWith(
      'PORTAL_SVCS_URL/MEM_SVC_CONTEXT/api/member/v1/members/byMemberCk/123456789',
      {
        cache: undefined,
        headers: { Authorization: 'Bearer BearerTokenMockedValue' },
        next: { revalidate: 1800, tags: ['123456789'] },
      },
    );
  });
});
