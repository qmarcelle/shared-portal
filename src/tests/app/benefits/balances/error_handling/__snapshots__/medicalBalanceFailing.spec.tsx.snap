// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Medical Balances Error Handling with Dental working should call Balances api and render error for Medical Balance, success for Dental 1`] = `
<div>
  <main
    class="flex flex-col justify-center items-center page"
  >
    <span
      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
    />
    <div
      class="flex flex-col app-content app-base-font-color"
    >
      <h1
        class="title-1 my-4 mb-0"
      >
        Balances
      </h1>
      <p
        class="body-1 body-1 my-4 mb-0 w-2/3"
      >
        Your balances show how close you are to meeting your annual plan deductibles and out-of-pocket limits.
      </p>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 9px; min-height: 9px;"
      />
      <section
        class="flex flex-row items-start app-body"
      >
        <div
          class="flex flex-col flex-grow page-section-63_33 items-stretch"
        >
          <div
            class="card-main large-section"
          >
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              >
                Medical & Pharmacy Balance
              </h2>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="card-main neutral container undefined"
              >
                <div
                  class="flex flex-row flex flex-row align-top m-4"
                >
                  <img
                    alt="alert"
                    class="icon"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                  />
                  <div
                    class="flex flex-row flex flex-col flex-grow"
                  >
                    <p
                      class="body-1"
                    >
                      There was a problem loading your information. Please try refreshing the page or returning to this page later.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-main large-section"
          >
            <div
              class="flex flex-col"
            >
              <h2
                class="title-2"
              >
                Dental Balance
              </h2>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row flex flex-row"
              >
                <p>
                  Member :
                </p>
                <span
                  style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                />
                <div
                  class="relative"
                >
                  <div
                    class="flex flex-row link"
                  >
                    <p
                      class="body-bold"
                    >
                      ChrisBalance HALL
                    </p>
                    <span
                      style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                    />
                    <img
                      alt="down icon"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <section
                    class="card-elevated z-20 dropdown-section"
                    style="display: none; position: absolute;"
                  >
                    <div
                      class="flex flex-row dropdown-item-selected dropdown-item p-2 cursor-pointer"
                    >
                      <div
                        class="size-5 mx-2 "
                      >
                        <img
                          alt="selcted"
                          class="size-5"
                          data-nimg="1"
                          decoding="async"
                          height="40"
                          loading="lazy"
                          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                          style="color: transparent;"
                          width="40"
                        />
                      </div>
                      <div>
                        <p
                          class="whitespace-nowrap mx-2"
                        >
                          ChrisBalance HALL
                        </p>
                      </div>
                    </div>
                    <div
                      class="flex flex-row  dropdown-item p-2 cursor-pointer"
                    >
                      <div
                        class="size-5 mx-2"
                      />
                      <div>
                        <p
                          class="whitespace-nowrap mx-2"
                        >
                          CHRISTMAS HALL
                        </p>
                      </div>
                    </div>
                  </section>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-col"
              >
                <p
                  class=" app-underline body-1"
                >
                  Deductible
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="card-main undefined"
                  style="background-color: rgba(0, 0, 0, 0.05);"
                >
                  <div
                    class="flex flex-col m-4"
                  >
                    <div
                      class="flex flex-row"
                    >
                      <p
                        class="body-1 body-1"
                      >
                        You do not have any Deductible amounts.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-col"
              >
                <p
                  class=" app-underline body-1"
                >
                  Out-of-Pocket
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="card-main undefined"
                  style="background-color: rgba(0, 0, 0, 0.05);"
                >
                  <div
                    class="flex flex-col m-4"
                  >
                    <div
                      class="flex flex-row"
                    >
                      <p
                        class="body-1 body-1"
                      >
                        You do not have any Out-of-Pocket amounts.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-col"
              >
                <p
                  class="underline decoration-dashed underline-offset-4 app-underline body-1"
                >
                  Services Used
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div>
                  <div
                    class="flex flex-row"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold"
                      >
                        $0.00
                      </p>
                      <p
                        class="body-2"
                      >
                        Used
                      </p>
                    </div>
                    <p
                      class="body-1 ml-5"
                    >
                      $3000 Annual Maximum Basic and Major
                    </p>
                  </div>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                  />
                  <div
                    class=""
                    style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                  />
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                  />
                </div>
                <div>
                  <div
                    class="flex flex-row"
                  >
                    <div
                      class="flex flex-col"
                    >
                      <p
                        class="body-1 font-bold"
                      >
                        $0.00
                      </p>
                      <p
                        class="body-2"
                      >
                        Used
                      </p>
                    </div>
                    <p
                      class="body-1 ml-5"
                    >
                      $3000 Ortho Lifetime Maximum
                    </p>
                  </div>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                  />
                  <div
                    class=""
                    style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                  />
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                  />
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                />
                <div
                  class="flex flex-col"
                >
                  <span
                    class="rich-text body-2 undefined"
                  >
                    <span>
                      Services Used is based on your processed items. There may be a delay in the Services Used list updating. If you're unsure if a service has been used,
                       
                    </span>
                    <span
                      class="link"
                    >
                      <a>
                         start a chat 
                      </a>
                    </span>
                    <span>
                       or call us at [1-800-000-0000].
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col  flex-grow page-section-36_67 items-stretch"
        >
          <div
            class="card-main large-section"
          >
            <div>
              <h2
                class="title-2"
              >
                Spending Summary
              </h2>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
              />
              <div
                class="flex flex-row"
              >
                <p
                  class="body-1"
                >
                  As of
                </p>
                <span
                  style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                />
                <p>
                  October 12, 2023
                </p>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
              />
              <div
                class="flex flex-col"
              >
                <div
                  class="flex flex-col doughnut-img mx-auto"
                >
                  <canvas
                    class="spendingSummaryChart"
                    height="150"
                    role="img"
                    width="300"
                  />
                  <div
                    class="doughnut-middle-text"
                  >
                    <div
                      class="flex flex-col justify-between pl-3 chartText "
                    >
                      <p>
                        Your plan paid
                      </p>
                      <p
                        class="title-2-bold"
                      >
                        89
                        %
                      </p>
                      <p>
                        of your costs
                      </p>
                    </div>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                    <div
                      class="chartTextDivider"
                    >
                      <div
                        class=""
                        style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                      />
                    </div>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                    <div
                      class="flex flex-col justify-between pl-3 chartText"
                    >
                      <p>
                        Total Billed
                      </p>
                      <p
                        class="title-2-bold"
                      >
                        $9,804.31
                      </p>
                    </div>
                  </div>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
              </div>
              <div
                class="sectionAlignment"
              >
                <div
                  class="flex flex-row"
                >
                  <div
                    style="width: 20px; height: 20px; background-color: rgb(0, 94, 185); border-radius: 50px; margin-top: 0px; margin-right: 0px;"
                  />
                  <p
                    class="body-1 ml-2"
                  >
                    You Paid
                  </p>
                  <p
                    class="body-bold ml-auto"
                  >
                    $1,199.19
                  </p>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
                />
                <div
                  class="flex flex-row"
                >
                  <div
                    style="width: 20px; height: 20px; background-color: rgb(93, 193, 253); border-radius: 50px; margin-top: 0px; margin-right: 0px;"
                  />
                  <p
                    class="body-1 ml-2"
                  >
                    You Saved
                  </p>
                  <p
                    class="body-bold ml-auto"
                  >
                    $8,605.12
                  </p>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <a
                style="display: block;"
                tabindex="-1"
              >
                <button
                  class="flex flex-row link-container undefined"
                  style="max-width: max-content; height: auto; display: block;"
                  tabindex="0"
                >
                  <p
                    class="link undefined"
                    style="display: block;"
                  >
                    View Spending Summary
                  </p>
                </button>
              </a>
              <div
                class="flex flex-col clear-left"
              />
            </div>
          </div>
          <div
            class="card-main large-section"
          >
            <div>
              <h1
                class="title-1 title-2"
              >
                Spending Accounts
              </h1>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row"
              >
                <div
                  class="flex flex-col"
                >
                  <p
                    class="body-1 py-4 px-7 font-bold"
                    style="background-color: rgb(231, 246, 255); border-radius: 7px;"
                  >
                    HSA
                  </p>
                </div>
                <div
                  class="flex flex-col ml-4"
                >
                  <p
                    class="body-2"
                  >
                    Health Saving Account Balance
                  </p>
                  <p
                    class="body-1 font-bold"
                  >
                    $349.90
                  </p>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 21px; min-height: 21px;"
              />
              <div
                class="flex flex-row"
              >
                <div
                  class="flex flex-col"
                >
                  <p
                    class="body-1 py-4 px-7 font-bold"
                    style="background-color: rgb(231, 246, 255); border-radius: 7px;"
                  >
                    FSA
                  </p>
                </div>
                <div
                  class="flex flex-col ml-4"
                >
                  <p
                    class="body-2"
                  >
                    Flexible Spending Account Balance
                  </p>
                  <p
                    class="body-1 font-bold"
                  >
                    $1,009.50
                  </p>
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
              />
              <a
                href=""
                style="display: block;"
                tabindex="-1"
              >
                <button
                  class="flex flex-row link-container undefined"
                  style="max-width: max-content; height: auto; display: block;"
                  tabindex="0"
                >
                  <p
                    class="link undefined"
                    style="display: block;"
                  >
                    View Spending Accounts
                  </p>
                </button>
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </main>
</div>
`;
